jsonschema-validation.xml

```diff
@@ -179,15 +179,26 @@
 
             <section title="Validation of primitive types and child values">
                 <t>
-                    Two of the primitive types, array and object, allow for child values.  The validation of
-                    the primitive type is considered separately from the validation of child instances.
+                    Two of the primitive types, array and object, allow for child values.
+                    The validation of the primitive type is considered separately from
+                    the validation of child instances.
                 </t>
                 <t>
-                    For arrays, primitive type validation consists of validating restrictions on length.
+                    For arrays, primitive type validation consists of validating
+                    restrictions on length with "minItems" and "maxItems", while
+                    "items" and "additionalItems" determine which subschemas apply
+                    to which elements of the array.  In addition, "uniqueItems"
+                    and "contains" validate array contents as a whole.
                 </t>
                 <t>
-                    For objects, primitive type validation consists of validating restrictions on the presence
-                    or absence of property names.
+                    For objects, primitive type validation consists of validating
+                    restrictions on which and how many properties appear with
+                    "required", "minProperties", "maxProperties", "propertyNames",
+                    and the string array form of "dependencies", while "properties",
+                    "patternProperties", and "additionalProperties" determine which
+                    subschemas apply to which object property values.
```

Comment:
@Relequestual I have made the changes you requested and replaced "validation always succeeds" with "does not directly validate the instance itself."